app/models/activity.rb

```diff
@@ -81,6 +96,7 @@ class Activity < ApplicationRecord
   FORM_STATE_VALIDATION_LIST = FORM_STEPS.map(&:to_s).push("complete")
 
   before_validation :strip_control_characters_from_fields!
+  before_save :set_non_oda_defaults, if: %i[is_oda_changed? is_non_oda?]
```

Comment:
Could this go wrong? I know one of the reasons some people avoid ActiveRecord callbacks is because they can cause unintended effects.